pkg/subctl/cmd/join.go

```diff
@@ -459,10 +482,8 @@ func populateSubmarinerSpec(subctlData *datafile.SubctlData) submariner.Submarin
 		Namespace:                SubmarinerNamespace,
 		CableDriver:              cableDriver,
 		ServiceDiscoveryEnabled:  subctlData.ServiceDiscovery,
+		GlobalCIDR:               globalCIDR,
```

Comment:
Keep it the way it was. We only fill the filled if it was configured else leave it at nil. If globalnet is not configured globalCIDR will be "" and we don't want an empty non-nil field. That is assuming CR treats nil and empty fields differently.